modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage/corrupted_tongue.dm

```diff
@@ -0,0 +1,141 @@
+/// Maximum an Hemophage will drain, they will drain less if they hit their cap.
+#define HEMOPHAGE_DRAIN_AMOUNT 50
+/// The multiplier for blood received by Hemophages out of humans with ckeys.
+#define BLOOD_DRAIN_MULTIPLIER_CKEY 1.5
+
+
+/datum/component/organ_corruption/tongue
+	corruptable_organ_type = /obj/item/organ/internal/tongue
+	corrupted_icon_state = "tongue"
+	/// The item action given to the tongue once it was corrupted.
+	var/tongue_action_type = /datum/action/cooldown/hemophage/drain_victim
+
+
+/datum/component/organ_corruption/tongue/corrupt_organ(obj/item/organ/internal/corruption_target)
+	. = ..()
+
+	if(!.)
+		return
+
+	var/obj/item/organ/internal/tongue/corrupted_tongue = corruption_target
+	corrupted_tongue.liked_foodtypes = BLOODY
+	corrupted_tongue.disliked_foodtypes = NONE
+
+	var/datum/action/tongue_action = corruption_target.add_item_action(tongue_action_type)
```

Comment:
> Since this is attached to an organ, which exists in nullspace, this proc won't actually be able to add the action to the mob. The proc is written with the assumption that the item exists in the mob's loc.

That's actually why I'm granting the action manually. It goes through the `actions` automatically during `on_insert()` to grant them to the mob, and actually removes them during `on_remove()`, specifically for organs.

This is actually 100% functional, and works exactly as intended.